### PR TITLE
bugfix/21098-ie11-compatibility-issues

### LINKED ISSUE #21098: Highcharts v11.3 doesn't work properly on internet explorer 11
We recently updated our application from highcharts v7 to the v11.3; since we need to be compatible with internet explorer 11, we load the library (highstock) from the _es5_ folder as shown here https://www.highcharts.com/docs/getting-started/system-requirements.
All the charts are drawn but we encountered the following problems:
1. There is no way to show the tooltip.
2. The event click doesn't work, we have to click several times on the chart (we especially tested pie charts) and only few times the event is fired.

Unfortunately I haven't found a fiddle that works with ie11 or edge in internet explorer mode.
Here there is an example you can try on internet explorer:

```
<html>
	<head>		
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
		<script language="javascript" type="text/javascript" src="https://code.highcharts.com/es5/highstock.js"></script>
		<script type="text/javascript"> 	
	$(function () {
		$('#container').highcharts({
		
			chart: {
				type: 'pie'
			},
			plotOptions: {
				series: { 
					cursor: 'pointer' 
				},
				pie: {  
					allowPointSelect: true, 
					cursor: 'pointer',
					dataLabels: {
						enabled: false
					},
					showInLegend: false,
					
					events: {
						click: function(event) {
							// event click function...
							alert ("clicked");
						}					
					}
				}
			},
			tooltip: {
				enabled: true
			},
			series: [
				{
					name: 'Percentage',
					colorByPoint: true,
					data: [
						{
							name: 'Water',
							y: 55.02
						},
						{
							name: 'Fat',                    
							y: 26.71
						},
						{
							name: 'Carbohydrates',
							y: 1.09
						},
						{
							name: 'Protein',
							y: 15.5
						},
						{
							name: 'Ash',
							y: 1.68
						}
					]
				}
			]
		});
		});

		</script>
	<head>
	<body>
		<div id="container">
		</div>
	</body>
</html>
```

### PR DESCRIPTION
**Pull request by Xillez**
Fixed #21098, added minor fixes for IE11 compatibility

-----

Asana task: https://app.asana.com/0/1204848917817812/1207247168714542/f

- Adds some more potential issues & workarounds related to IE11 compatibility to system-requirements.md
- Minor fixes to how inClass handles parental traversing.
- Removed deprecated dissolve call in highcharts.src.ts

### CODE REVIEW

File: docs/getting-started/system-requirements.md [hunk 1]
```diff
@@ -31,3 +31,103 @@ Beginning with Highcharts v11, legacy browsers must load Highcharts from the `es
...
+            }
+
+            CustomEvent.prototype = window.Event.prototype;
+
+            window.CustomEvent = CustomEvent;
+        })();
+
+    </script>
+
+    <script src="http://localhost:3031/es5/highcharts.src.js"></script>
```
Comment (bre1470) [inline docs/getting-started/system-requirements.md:98]: ES5-variant of `highcharts.src.js` includes the polyfills. If it is not working in IE11, the `ts/masters-es5/highcharts.src.ts` file has to be improved.

File: ts/Core/Pointer.ts [hunk 2]
```diff
@@ -918,7 +919,19 @@ class Pointer {
...
+                elem && (
+                    // HTMLElement
+                    elem === document.documentElement ||
+                    // Document
+                    defined(elem.nodeType) &&
+                    elem.nodeType === document.nodeType
+                )
+            ) {
+                elem = null;
+            }
```
Comment (bre1470) [inline ts/Core/Pointer.ts:934]: This should go into the MSPointer class by overwriting `Pointer.inClass`.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Comment by bre1470** [comment]
You are on the right track, just the places of adjustment need a change.

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Review by KacperMadej — APPROVED** [review]
LGTM 👍

File: docs/getting-started/system-requirements.md [hunk 3]
```diff
@@ -31,3 +31,36 @@ Beginning with Highcharts v11, legacy browsers must load Highcharts from the `es
...
+
+    ```css
+    .highcharts-tooltip > span {
+        background: rgba(255, 255, 255, 0.85);
+        border: 1px solid silver;
+        border-radius: 3px;
+        box-shadow: 1px 1px 2px #888;
+        padding: 8px;
+    }
+    ```
```
Comment (TorsteinHonsi) [inline docs/getting-started/system-requirements.md:66]: Instead of offering a workaround in the docs, I think it is better if we just make it work by means of a graceful degradation. We can add a snippet in the polyfills.ts or some other appropriate place to disable the shadow for IE11, then we save the clients for the hassle of debugging.

Something like this: https://jsfiddle.net/highcharts/bu7t4hgo/

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
